### DISCUSSION TITLE
elevate points above ground based on Z value

### DISCUSSION THREAD

**Discussion by wiesehahn** [maplibre/maplibre-gl-js#3397] [General]
I just saw an example where points where converted to polygons to elevate them above terrain using a workaround with fill extrusion and base height. 
I was wondering if there is a way to directly draw elevated points on a map? 

Points can have Z values defining the height, it would be nice to be able to vizualize these points elevated above ground if the map is pitched. 

[![grafik](https://github.com/maplibre/maplibre-gl-js/assets/41429613/45975c0b-362e-4160-9e6a-0dea4b488ec7)](https://twitter.com/i/status/1726834928808034348)

**Comment by HarelM** [comment]
I think this is similar to this proposal to some extent:
https://github.com/maplibre/maplibre-style-spec/issues/62
The problem with elevating a point is that you need a 3D representation of that point - it is a circle? a square? what is its dimensions? etc, which are currently not covered in the spec.
For fill extrusion this is defined better and therefore you can elevate those.
I think you can also do it using a custom layer as you have a way to interact with the gl context, so that's another option, both options are not straight forwards obviously.

This dome is nice :-)
